tests/model_inheritance/tests.py

```diff
@@ -399,48 +398,37 @@ def test_exclude_inherited_on_null(self):
         )
 
 
-class InheritanceSameModelNameTests(TransactionTestCase):
+@isolate_apps('model_inheritance', 'model_inheritance.tests')
+class InheritanceSameModelNameTests(SimpleTestCase):
+    def test_abstract_fk_related_name(self):
+        class Referenced(models.Model):
+            class Meta:
+                app_label = 'model_inheritance'
 
-    available_apps = ['model_inheritance']
+        class AbstractReferent(models.Model):
+            reference = models.ForeignKey(
+                Referenced, models.CASCADE, related_name='%(app_label)s_%(class)s_references'
```

Comment:
I'd use a local variable for '%(app_label)s_%(class)s_references' so we don't risk typoing it in the assertion.